### PR TITLE
Extract View and Projection definition

### PR DESCRIPTION
**Pull request by nechaido**

### CODE REVIEW

File: lib/schema.js [hunk 1]
```diff
@@ -129,6 +129,48 @@ const linkSchemas = (schemas, categories, domains) => {
...
+    for (const key in props) {
+      const prop = props[key];
+      if (typeof prop !== 'string') {
+        view.definition[key] = prop;
+        continue;
+      }
+      let schema = ms.categories.get(view.Category);
+      if (!schema) {
+        schema = ms.views.get(view.Category);
+      }
```
Comment (lundibundi) [inline lib/schema.js]: Nit: 
```js
const schema =
 ms.categories.get(view.Category) || ms.views.get(view.Category);
```

File: test/schemas/view/FullName.schema [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    FirstName: { domain: 'Nomen' },
+    LastName: { domain: 'Nomen' },
+}
```
Comment (lundibundi) [inline test/schemas/view/FullName.schema]: I think the indentation in schemas should be 2, isn't it?

File: lib/schema.js [hunk 3]
```diff
@@ -129,6 +129,48 @@ const linkSchemas = (schemas, categories, domains) => {
...
+};
+
+const processView = (view, ms) => {
+  if (view.definition) {
+    return;
+  }
+  view.definition = {};
+  for (const field of view.Fields) {
+    const props = typeof field === 'string' ? { [field]: field } : field;
+    for (const key in props) {
```
Comment (lundibundi) [inline lib/schema.js]: Nit: perhaps `Object.keys(props)[0]` to make sure we only support one key there or signify that there shouldn't be more than one field.

Comment (nechaido) [reply lib/schema.js]: This is intentionally implemented this way to support:
```js
View({
  Category: 'Name',
  Fields: [{
    A: 'B',
    B: 'A',
  }],
})
```

File: test/schemas/view/FullName.schema (hunk 2 above)
Comment (nechaido) [reply test/schemas/view/FullName.schema]: AFAIK it's not specified anywhere, we have no linting rules for schemas,  and indentation has no effect on it's parsing.

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/metaschema/commit/21a4e0a1ae50267958777dc7a7d23203d3af9db1.

**Pull request closed without merging**